extensions/wikia/PageShare/PageShareHelper.class.php

```diff
@@ -23,46 +22,41 @@ public static function getIcon( $service ) {
 
 	/**
 	 * Get language for Page Share service.
-	 * For anon users use the browser language from client side, if empty default to EN.
-	 * For logged in user use user's language.
-	 * Both values can be overwritten by ?uselang parameter which is passed from the client side.
+	 * Language code is sent from the client side. If it's empty, the default (en) is returned.
 	 *
-	 * @param $browserLang
-	 * @param $useLang
+	 * @param $requestShareLang
 	 * @return String language
 	 */
-	public static function getLangForPageShare( $browserLang, $useLang ) {
-		global $wgLang, $wgUser;
-
-		if ( !empty ( $useLang ) ) {
-			return $useLang;
-		} else if ( $wgUser->isAnon() ) {
-			if ( !empty( $browserLang ) ) {
-				return $browserLang;
-			} else {
-				return self::SHARE_DEFAULT_LANGUAGE;
-			}
-		} else {
-			return $wgLang->getCode();
-		}
+	public static function getLangForPageShare( $requestShareLang ) {
+		return empty( $requestShareLang ) ? self::SHARE_DEFAULT_LANGUAGE : $requestShareLang;
 	}
 
 	public static function isValidShareService( $service, $lang, $isTouchScreen ) {
-		// Don't display LINE social network (because it's a mobile app) on desktops
-		if ( !$isTouchScreen && $service['name'] === self::LINE_NAME ) {
+		// Don't display LINE social network or any other mobile app on desktops
+		if (
+			!$isTouchScreen
+			&& array_key_exists( 'displayOnlyOnTouchDevices', $service )
+			&& $service['displayOnlyOnTouchDevices']
```

Comment:
I'm wondering if it is possible to use this function https://github.com/Wikia/app/blob/0d3b259720799abb42e9cc281d4faef65ffda12c/includes/OutputPage.php#L1817 → we wouldn't need to pass anything, just read the headers on the API call.